omxplayer/event.py

```diff
@@ -0,0 +1,80 @@
+import logging
+logger = logging.getLogger(__name__)
+
+class Event:
+    def __init__(self):
+        self.handlers = set()
+        self.counter = 0
+        self.fireStack = 0
+        self._handle_queue = []
+        self._unhandle_queue = []
+
+    def handle(self, handler):
+        if self.isFiring():
+            self._handle_queue.append(handler)
+            return self
+
+        self.handlers.add(handler)
+        return self
+
+    def unhandle(self, handler):
+        if self.isFiring():
+            self._unhandle_queue.append(handler)
+            return self
+
+        try:
+            self.handlers.remove(handler)
+        except:
+            # raise ValueError("Handler is not handling this event, so cannot unhandle it.")
+            logger.warning('Event.unhandle got unknown handler')
+            # traceback.print_exc()
+            # traceback.print_stack()
+
+        return self
+
+    def handles(self, handler):
+        return handler in self.handlers
+
+    def fire(self, *args, **kargs):
+        # change state
```

Comment:
This comment doesn't add anything, what are we changing state from to? Ideally we should express this in meaning variable names and method calls rather than comments.